infra/af-mvp/resources/domainSetup.ts

```diff
@@ -38,16 +28,50 @@ export function createDomainSetup() {
       { protect: true } // Keep the zone from being destroyed
     );
 
-    const { certificate } = createDomainRecordAndCertificate(
-      zone,
-      domainConfig.domainName,
-      cdnDomainName
-    );
+    const cdnURL = await currentStackReference.getOutputValue('cdnURL');
+    if (!cdnURL) {
+      console.log(
+        "Skipped creating domain configurations as there's a circular dependency to the CDN which is not yet created."
+      );
+      return;
+    }
+
+    const cdnDomainName = parseDomainNameFromURL(cdnURL);
+    let domainCertificate: aws.acm.Certificate;
+
+    if (domainConfig.domainName === domainConfig.domainRootName) {
+      // If the domain name is the same as the zone name, we cant' create a CNAME record for as it would conflict with the zone apex record.
+      // We create ALIAS record instead.
+      new aws.route53.Record(nameResource('domainAliasRecord'), {
```

Comment:
Pitäisköhän tässä kommentissa implikoida, että tämä liittyy staging/prod subdomain hallintaan(?). Tai näin ainakin ymmärsin, piti vähän aikaa ihmetellä, että miksi tässä näin iffitellään, mutta ilmeisesti siis prod -tapauksessa domainName ja domainRootName on samat, staging käyttelee sit subdomainia.

Reply:
Joo hyvä huomio! Selventelen tuota esimerkillä. Sinäänsä ei teknisesti liity eri live-ympäristöihin vaan pelkästään siihen minkälainen domain-nimi on kyseessä (eli prod voisi yhtä hyvin olla sellainen jossa domain-nimi on alidomain, tai vastaavasti staging vois olla juuridomain.).

Reply:
> Oliko tämä staging.accessfinland.dev jokin sovittu juttu, että pitää käytellä?
Juu oli, ja tarkoitus taisi olla sellainen että voisi myöhemmin löytyä jotain vaikka `demo.accessfinland.dev` -tyyppistä käyttöä.